tests/tests/proposal_tests.cpp

```diff
@@ -0,0 +1,116 @@
+ */
+
+#include <boost/test/unit_test.hpp>
+
+#include <graphene/chain/exceptions.hpp>
+#include <graphene/chain/account_object.hpp>
+#include <graphene/chain/proposal_object.hpp>
+
+#include <fc/crypto/digest.hpp>
+
+#include "../common/database_fixture.hpp"
+
+using namespace graphene::chain;
+using namespace graphene::chain::test;
+
+BOOST_FIXTURE_TEST_SUITE( proposal_tests, database_fixture )
+
+BOOST_AUTO_TEST_CASE( proposal_failure )
+{
+   try
+   {
+      // create an account for Bob
+      fc::ecc::private_key bob_key = fc::ecc::private_key::regenerate(fc::digest("bobkey"));
+      const account_object& bob = create_account( "bob", bob_key.get_public_key() );
+      const account_id_type bob_id = bob.get_id();
+      fund( bob, asset(1000000) );
+
+      // add an account for Alice
+      fc::ecc::private_key alice_key = fc::ecc::private_key::regenerate(fc::digest("alicekey"));
+      const account_object& alice = create_account( "alice", alice_key.get_public_key() );
+      const account_id_type alice_id = alice.get_id();
+      fund(alice, asset(1000000) );
+
+      set_expiration( db, trx );
+
+      // create proposal that will eventually fail due to lack of funds
+      transfer_operation top;
+      top.to = alice_id;
+      top.from = bob_id;
+      top.amount = asset(2000000);
+      proposal_create_operation pop;
+      pop.proposed_ops.push_back( { top } );
+      pop.expiration_time = db.head_block_time() + fc::days(1);
+      pop.fee_paying_account = bob_id;
+      trx.operations.push_back( pop );
+      trx.signatures.clear();
+      sign( trx, bob_key );
+      proposal_object prop;
+
+      try
+      {
+         processed_transaction processed = PUSH_TX( db, trx );
+         prop = db.get<proposal_object>(processed.operation_results.front().get<object_id_type>());
+      }
+      catch( fc::exception& ex )
+      {
+         BOOST_FAIL( "Unable to create proposal. Reason: " + ex.to_string(fc::log_level(fc::log_level::all)));
+      }
+
+      trx.clear();
+      generate_block();
+
+      // make sure proposal is still there
+      try {
+         db.get<proposal_object>(prop.id);
+      }
+      catch (fc::exception& ex)
+      {
+         BOOST_FAIL( "proposal object no longer exists after 1 block" );
+      }
+
+      // add signature
+      proposal_update_operation up_op;
+      up_op.proposal = prop.id;
+      up_op.fee_paying_account = bob_id;
+      up_op.active_approvals_to_add.emplace( bob_id );
+      trx.operations.push_back( up_op );
+      sign( trx, bob_key );
+      PUSH_TX( db, trx );
+      trx.clear();
+
+      // check fail reason
+      const proposal_object& result = db.get<proposal_object>(prop.id);
+      BOOST_CHECK_EQUAL(result.fail_reason, "Assert Exception: insufficient_balance: Insufficient Balance: "
+            "10 BTS, unable to transfer '20 BTS' from account 'bob' to 'alice' Unable to transfer 20 BTS "
+            "from bob to alice");
+   }
+   catch (fc::exception& ex)
```

Comment:
Please use FC_LOG_AND_RETHROW macro instead.

Reply:
Fixed in next check in.